contracts/modules/cover/Cover.sol

```diff
@@ -638,64 +639,44 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
-    uint[] calldata productTypeIds,
-    uint16[] calldata gracePeriodsInDays,
-    string[] calldata ipfsMetadata
-  ) external override onlyAdvisoryBoard {
-
-    for (uint i = 0; i < productTypeIds.length; i++) {
-      _productTypes[productTypeIds[i]].gracePeriodInDays = gracePeriodsInDays[i];
-      emit ProductTypeSet(productTypeIds[i], ipfsMetadata[i]);
+    for (uint i = 0; i < productParams.length; i++) {
+      ProductParam calldata param = productParams[i];
+      validateProduct(param.product, _coverAssetsFallback, productTypesCount);
+      // New product has id == uint256.max
+      if (param.productId == type(uint256).max) {
+        emit ProductSet(_products.length, param.ipfsMetadata);
+        _products.push(param.product);
+      } else {
+        // existing product
+        require(param.productId < _products.length, "Cover: Product doesnt exist. Set id to uint256.max to add it");
+        Product storage newProductValue = _products[param.productId];
+        newProductValue.isDeprecated = param.product.isDeprecated;
+        newProductValue.coverAssets = param.product.coverAssets;
+        newProductValue.initialPriceRatio = param.product.initialPriceRatio;
+        newProductValue.capacityReductionRatio = param.product.capacityReductionRatio;
+        emit ProductSet(param.productId, param.ipfsMetadata);
+      }
     }
   }
 
-  function editProducts(
-    uint[] calldata productIds,
-    ProductUpdate[] calldata productUpdates,
-    string[] calldata ipfsMetadata
-  ) external override onlyAdvisoryBoard {
-
-    uint32 _coverAssetsFallback = coverAssetsFallback;
-    uint productTypesCount = _productTypes.length;
-
-    for (uint i = 0; i < productIds.length; i++) {
+  function setProductTypes(ProductTypeParam[] calldata productTypeParams) external onlyAdvisoryBoard {
 
-      Product memory newProductValue = _products[productIds[i]];
-      newProductValue.coverAssets = productUpdates[i].coverAssets;
-      newProductValue.initialPriceRatio = productUpdates[i].initialPriceRatio;
-      newProductValue.capacityReductionRatio = productUpdates[i].capacityReductionRatio;
+    for (uint i = 0; i < productTypeParams.length; i++) {
+      ProductTypeParam calldata param = productTypeParams[i];
 
-      validateProduct(newProductValue, _coverAssetsFallback, productTypesCount);
-      _products[productIds[i]] = newProductValue;
-      emit ProductSet(productIds[i], ipfsMetadata[i]);
+      // New product has id == uint256.max
+      if (param.productTypeId == type(uint256).max) {
+        emit ProductTypeSet(_productTypes.length, param.ipfsMetadata);
+        _productTypes.push(param.productType);
+      } else {
+        require(param.productTypeId < _productTypes.length, "Cover: ProductType doesnt exist. Set id to uint256.max to add it");
+        _productTypes[param.productTypeId].gracePeriodInDays = param.productType.gracePeriodInDays;
+        emit ProductTypeSet(param.productTypeId, param.ipfsMetadata);
```

Comment:
Same comments as for the products: add `continue` to drop an indent level and allow empty metadata